### PR TITLE
Fixing some code smells

### PR DESCRIPTION
**Pull request by pauarge**
## Description

This PR addresses some code smells detected by Sonarlint in the plugin's codebase. Each change has been commented with its reason.

We haven't tackled all issues in the code, we're being conservative with the changes.

## Motivation and Context

General code cleanliness.

## How Has This Been Tested?

Visual inspection of the plugin's functionality plus CI.

## Types of changes

Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: src/class-parsely.php [hunk 1]
```diff
@@ -1564,20 +1563,6 @@ private function get_logo_default(): string {
...
-	 * @return string $url The host of the url…
-	 */
-	private function get_host_from_url( $url ): string {
-		if ( preg_match( '/^https?:\/\/( [^\/]+ )\/.*$/', $url, $matches ) ) {
-			return $matches[1];
-		}
-
-		return $url;
-	}
-
```
Comment (pauarge) [inline src/class-parsely.php:1544]: Unused function

File: src/class-parsely.php [hunk 2]
```diff
@@ -946,7 +946,6 @@ public function construct_parsely_metadata( array $parsely_options, $post ): arr
 		} elseif ( in_array( get_post_type( $post ), $parsely_options['track_post_types'], true ) && self::post_has_trackable_status( $post ) ) {
 			$authors  = $this->get_author_names( $post );
 			$category = $this->get_category_name( $post, $parsely_options );
-			$post_id  = $parsely_options['content_id_prefix'] . get_the_ID();
```
Comment (pauarge) [inline src/class-parsely.php:949]: Unused variable

File: src/class-parsely-recommended-widget.php [hunk 3]
```diff
@@ -87,8 +87,7 @@ public function widget( $args, $instance ): void {
 		$removed_title_esc = remove_filter( 'widget_title', 'esc_html' );
 
 		/** This filter is documented in wp-includes/widgets/class-wp-widget-pages.php */
-		$title      = apply_filters( 'widget_title', $instance['title'] );
-		$title_html = $args['before_widget'] . $args['before_title'] . $title . $args['after_title'];
```
Comment (pauarge) [inline src/class-parsely-recommended-widget.php:91]: Duplicate `$title_html` declaration

Comment (GaryJones) [reply src/class-parsely-recommended-widget.php:91]: Nice catch.

**Review by GaryJones — APPROVED** [review]
✅

**Pull request merged**